Source/control.cpp

```diff
@@ -1622,56 +1617,22 @@ void CheckBtnUp()
 
 void FreeControlPan()
 {
-	void *ptr;
-
-	ptr = pBtmBuff;
-	pBtmBuff = NULL;
-	mem_free_dbg(ptr);
-	ptr = pManaBuff;
-	pManaBuff = NULL;
-	mem_free_dbg(ptr);
-	ptr = pLifeBuff;
-	pLifeBuff = NULL;
-	mem_free_dbg(ptr);
-	ptr = pPanelText;
-	pPanelText = NULL;
-	mem_free_dbg(ptr);
-	ptr = pChrPanel;
-	pChrPanel = NULL;
-	mem_free_dbg(ptr);
-	ptr = pSpellCels;
-	pSpellCels = NULL;
-	mem_free_dbg(ptr);
-	ptr = pPanelButtons;
-	pPanelButtons = NULL;
-	mem_free_dbg(ptr);
-	ptr = pMultiBtns;
-	pMultiBtns = NULL;
-	mem_free_dbg(ptr);
-	ptr = pTalkBtns;
-	pTalkBtns = NULL;
-	mem_free_dbg(ptr);
-	ptr = pChrButtons;
-	pChrButtons = NULL;
-	mem_free_dbg(ptr);
-	ptr = pDurIcons;
-	pDurIcons = NULL;
-	mem_free_dbg(ptr);
-	ptr = pQLogCel;
-	pQLogCel = NULL;
-	mem_free_dbg(ptr);
-	ptr = pSpellBkCel;
-	pSpellBkCel = NULL;
-	mem_free_dbg(ptr);
-	ptr = pSBkBtnCel;
-	pSBkBtnCel = NULL;
-	mem_free_dbg(ptr);
-	ptr = pSBkIconCels;
-	pSBkIconCels = NULL;
-	mem_free_dbg(ptr);
-	ptr = pGBoxBuff;
-	pGBoxBuff = NULL;
-	mem_free_dbg(ptr);
+	MemFreeDbg(pBtmBuff);

```

Comment:
Is `FreeControlPan` bin exact with this change? I thought it also had to write `NULL` to every global variable, e.g.

```c
pBtmBuff = NULL
```

Reply:
Yes. `MemFreeDbg` is a macro in `defs.h` that does just that! Line numbers match up with the debug this way.